### PR TITLE
Add trace logging to transport

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Added trace logging to transport and enabled it for the `AtomixTransportTest`. I hope with that we can better understand why the test is sometimes failing.

## Related issues

related #4911 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to backport the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the release annoncement

### CODE REVIEW

File: transport/src/main/java/io/zeebe/transport/impl/AtomixClientTransportAdapter.java [hunk 1]
```diff
@@ -111,6 +126,12 @@ private void tryToSend(final RequestContext requestContext) {
       return;
     }
 
+    LOG.trace(
```
Comment (pihme) [inline transport/src/main/java/io/zeebe/transport/impl/AtomixClientTransportAdapter.java:129]: Maybe wrap in `isTraceEnabled()`

File: transport/src/main/java/io/zeebe/transport/impl/AtomixClientTransportAdapter.java [hunk 2]
```diff
@@ -122,14 +143,21 @@ private void tryToSend(final RequestContext requestContext) {
   private void handleResponse(
       final RequestContext requestContext, final byte[] response, final Throwable errorOnRequest) {
     if (requestContext.isDone()) {
+      LOG.trace("Handle response, but request {} is already done", requestContext.hashCode());
```
Comment (pihme) [inline transport/src/main/java/io/zeebe/transport/impl/AtomixClientTransportAdapter.java:146]: Maybe wrap in `isTraceEnabled()`. Also at other places in this class

File: transport/src/main/java/io/zeebe/transport/impl/AtomixServerTransport.java [hunk 3]
```diff
@@ -68,9 +65,11 @@ public void close() {
   public ActorFuture<Void> subscribe(final int partitionId, final RequestHandler requestHandler) {
     return actor.call(
         () -> {
+          final var topicName = topicName(partitionId);
+          LOG.trace("Subscribe for topic {}", topicName);
```
Comment (pihme) [inline transport/src/main/java/io/zeebe/transport/impl/AtomixServerTransport.java:69]: Maybe wrp in `ìsTraceEnabled()` also at other places in this class

**Review by pihme — APPROVED** [review]
Some comments; feel free to address or ignore them

File: transport/src/main/java/io/zeebe/transport/impl/AtomixServerTransport.java (hunk 3 above)
Comment (Zelldon) [reply transport/src/main/java/io/zeebe/transport/impl/AtomixServerTransport.java:69]: actually this is done by ZbLogger as far as I know

Comment (pihme) [reply transport/src/main/java/io/zeebe/transport/impl/AtomixServerTransport.java:69]: No, it is not. ZBLogger will prevent it from being logged (which ordinary loggers do anyways), but still the string will need to be composed, which can be expensive.

In newer logging implementation you often find lambdas:
`LOG.trace(() -> .calculateLongString())` which circumvent this once and for all. But these methods aren't available in `Logger`

Comment (Zelldon) [reply transport/src/main/java/io/zeebe/transport/impl/AtomixServerTransport.java:69]: ok get what you mean thanks

**Comment by Zelldon** [comment]
@pihme I updated the branch

**Review by pihme — APPROVED** [review]

**Pull request merged**